protocol/dubbo/dubbo_invoker.go

```diff
@@ -87,15 +93,24 @@ func (di *DubboInvoker) Invoke(ctx context.Context, invocation protocol.Invocati
 		err    error
 		result protocol.RPCResult
 	)
-	if atomic.LoadInt64(&di.reqNum) < 0 {
+	if !di.BaseInvoker.IsAvailable() {
```

Comment:
This check is duplicated to below. What is it intended for?

Reply:
It's double-check pattern, I think.

Reply:
yes.